nsq/async.py

```diff
@@ -241,7 +251,7 @@ def close(self):
     def _read_size(self, data):
         try:
             size = struct.unpack('>l', data)[0]
-            self.stream.read_bytes(size, self._read_body)
+            self._read_bytes(size, self._read_body)
```

Comment:
Good point. Should _start_read() also call self.close() on exceptions? If so, we can just add self.close() to the _read_bytes exception handler.